File: backend/app/Models/MenuItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class MenuItem extends Model
{
    protected $fillable = [
        'category_id',
        'name',
        'nepali_name',
        'description',
        'image',
        'price',
        'is_available',
        'is_taxable',
        'is_service_chargeable',
        'dietary_tags',
        'spicy_level',
        'preparation_station',
    ];

    protected $casts = [
        'is_available' => 'boolean',
        'dietary_tags' => 'array',
        'price' => 'decimal:2',
    ];

    public function category(): BelongsTo
    {
        return $this->belongsTo(Category::class);
    }

    public function variants(): HasMany
    {
        return $this->hasMany(ItemVariant::class);
    }

    public function addons(): HasMany
    {
        return $this->hasMany(ItemAddon::class);
    }
}

File: frontend/lib/models/menu_item.dart
class MenuItem {
  final int id;
  final int categoryId;
  final String name;
  final String? nepaliName;
  final String? description;
  final String? image;
  final double price;
  final bool isAvailable;
  final List<String> dietaryTags;

  MenuItem({
    required this.id,
    required this.categoryId,
    required this.name,
    this.nepaliName,
    this.description,
    this.image,
    required this.price,
    required this.isAvailable,
    required this.dietaryTags,
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) {
    return MenuItem(
      id: json['id'],
      categoryId: json['category_id'],
      name: json['name'],
      nepaliName: json['nepali_name'],
      description: json['description'],
      image: json['image'],
      price: double.parse(json['price'].toString()),
      isAvailable: json['is_available'] == 1 || json['is_available'] == true,
      dietaryTags:
          (json['dietary_tags'] as List?)?.map((e) => e.toString()).toList() ??
          [],
    );
  }
}
